### PR TITLE
Support tiebreak based on length of original text.

### PR DESCRIPTION
**Pull request by aldhsu**
Hi @lotabout thanks for skim I'm enjoying it.

One thing I have wished for is tiebreaking based on the length of the original text, in my case shorter strings are preferred. This lets me select longer strings by continuing to type in Vim rather than use the arrow keys I have included an example below of the default behaviour of skim. In this example lets say we are selecting the `build_script_mod` directory by typing `mod`.

Before, just `sk`:
Screen Shot 2019-09-23 at 10 57 20 pm

After, running `sk --tiebreak=score,length`:
Screen Shot 2019-09-23 at 11 09 17 pm

This may accidentally help with https://github.com/lotabout/skim/issues/213.

### CODE REVIEW

**Comment by aldhsu** [comment]
I'm not sure about the test failure, it doesn't look like code that should be affected.

**Comment by zsugabubus** [comment]
Could you explain me what's the relation between length and "select [...] strings by continuing to type"?

**Comment by aldhsu** [comment]
@zsugabubus The folder structures I am working files like `module.ext` and `module/submodule.ext` are common. I should qualify I mostly use Skim in Vim. 

To give a concrete example, let's say you are searching for `bar.rs` and you have two files `bar.rs` and `bar/foo.rs`.
I haven't verified these examples work exactly as I say. There are nuances to the algorithm such as boundaries etc. This is just a demonstration of the behaviour I was seeing.

### Default skim
1. Type `bar`
2. `bar/foo.rs` has equal score as `bar.rs` and may appear first
3. To select `bar.rs` use arrow keys to move up one and select

### Skim and sort by length prefer shorter
1. Type `bar`
2. `bar.rs` will always appear first

To answer your question: 

> what's the relation between length and "select [...] strings by continuing to type"?

I'm not sure I really thought it through for more complex cases but for the example I have described above it makes sense that longer pathnames have access to more characters  to get the match you are looking for("continuing to type") eg. to select `bar/foo.rs` you could type `barf`. There is no extra qualification available  for `bar.rs` if you have typed `bar` and it isn't the top result.

I have been building Skim from my branch and linking it for the Vim plugin with `export SKIM_DEFAULT_OPTIONS="--tiebreak=score,length"
`, in my experience it has been working as I have wanted.

**Comment by zsugabubus** [comment]
Just asked, because what come into my mind is that length has absolutely nothing to do where the match was be found. It maybe works for your special directory structure but not for the general usecase. Or maybe I just misunderstood your purpose of the feature from the description.

*Case 1*
```
> skim
src/skim/main.rs
this/is/the/line/i/want/skim
```

*Case 2*
```
> aaa
aaaxxxxx
xxxxxaaa
xxxaaaxx
```

> I'm not sure I really thought it through for more complex cases but for the example I have described above it makes sense that longer pathnames have access to more characters to get the match you are looking for("continuing to type") 

Looking at your second picture (for me from your description) it would be more reasonable to sort "mod$"s first and just after them "src/model.rs" because if you continue typing you will lose "mod$"s but you can type "...el" and still get "src/model.rs". And it has nothing to do with length because whether you can continue or not typing is only depends on the place of the match.

**Comment by aldhsu** [comment]
Ah yes, that makes more sense! I'll try that out.

**Comment by aldhsu** [comment]
Another reason that `fzf` doesn't have an alphabetical sort either could be the overhead of comparing lengths is much lower than comparing strings alphabetically.

File: src/selection.rs [hunk 1]
```diff
@@ -520,6 +520,20 @@ fn build_compare_function(criterion: Vec<RankCriteria>) -> CompareFunction<Match
                         return b.rank.score.cmp(&a.rank.score);
                     }
                 }
+                RankCriteria::Length => {
+                    if a.item.orig_text.len() == b.item.orig_text.len() {
```
Comment (lotabout) [inline src/selection.rs]: @aldhsu `orig_text` is the text we got before transformation and ansi-parsing. So here we should use `get_text` or `get_text_struct` to fetch the displayed content.

File: src/item.rs [hunk 2]
```diff
@@ -27,7 +27,7 @@ pub struct Item {
     index: (usize, usize),
 
     // The text that will be ouptut when user press `enter`
-    orig_text: String,
+    pub orig_text: String,
```
Comment (lotabout) [inline src/item.rs]: @aldhsu Also personally I don't think it a good idea to use `pub` in most cases.

**Pull request merged**